### PR TITLE
feat(announcement): add `<rh-announcement>` v2

### LINKED ISSUE #1576: [feat] `<rh-announcement>`element
### Description

Create an Announcement banner element with the primary use for promotions at the top of a website. 

### Acceptance Criteria

- [x] Design done
- [x] Development done

### Image

Screenshot 2024-05-22 at 1 55 33 PM

### Link to design doc

https://www.figma.com/design/0z0OyR7NHF90LPyjT3y4XX/Personalization-patterns-v2?node-id=0-1&t=trKI2WECpLXlls8i-0

### Other resources

_No response_

### PR DESCRIPTION
**Pull request by adamjohnson**
## What I did

1. Moved `<rh-announcement>` to a new PR because the old PR (#1722) had so many merge conflicts.
1. Created `<rh-announcement>`. See the [Figma file](https://www.figma.com/design/0z0OyR7NHF90LPyjT3y4XX/Personalization-patterns-subsystem?node-id=1-2&m=dev).
1. Changed `imgleft` boolean attribute to `image-position="inline-start"`.
1. Added a lightdom-shim.css file for decreased FOUC

## Testing Instructions

1. Be sure to get up to speed on all the previous work done in #1722
1. View each of the demos in the DP:
    * [rh-announcement](https://deploy-preview-2153--red-hat-design-system.netlify.app/elements/announcement/demo/)
    * [Image position](https://deploy-preview-2153--red-hat-design-system.netlify.app/elements/announcement/demo/image-position/) - `inline-start` and `block-start`
    * [Events](https://deploy-preview-2153--red-hat-design-system.netlify.app/elements/announcement/demo/events/)
    * [Dismissible](https://deploy-preview-2153--red-hat-design-system.netlify.app/elements/announcement/demo/dismissible/)
    * [Color context](https://deploy-preview-2153--red-hat-design-system.netlify.app/elements/announcement/demo/color-context/)
1. Compare demos to be sure they match the Figma
1. Make sure everything is accessible
1. Make sure this component can be customized as needed
1. Slot in a real image, text, and CTA
1. Test with lots of / too much content in the `body` slot

## Notes to Reviewers

Closes #1576.

## To do's

- [x] Docs

### CODE REVIEW

**Comment by alypilkons** [comment]
@adamjohnson Looks great! One thing Dustin & I noticed - the image looks off center on mobile when the X is present.

**Comment by dmazon-redhat** [comment]
Comments from my review of the demos. Thanks Adam for sharing! 

### Spacing adjustments

#### Announcement documentation spacing for reference:
Screenshot 2025-02-04 at 9 14 18 PM

#### Space between the image and paragraph text should be 32px on desktop instead of 24px.
Screenshot 2025-02-04 at 9 13 59 PM

#### I noticed that there is extra margin of 32px on the right hand side after the default CTA which slightly makes the content not centered. This could be removed at a desktop size but we will want to make sure that at all breakpoint sizes there is at least a 16px buffer between the CTA / paragraph text and the close button.
Screenshot 2025-02-04 at 9 16 12 PM

#### The 32px margin on the right really impacts the mobile size as seen below. We can free up at least 16px of space to give the content some more room to stretch out.
Screenshot 2025-02-04 at 9 19 57 PM

#### Maybe applying 16px to the close button instead of the end of the CTA would cover the needed right spacing of 16px at mobile sizes
Screenshot 2025-02-04 at 9 24 46 PM

### Close button touch target

This might not be an issue but wanted to inquire about the touch target size for the close button. Does it at minimum meet the required 24x24px target size? When I look at the focus size it shows 16x16px Just trying to meet our accessibility standards for a really small icon that has a big ol' functional purpose for users.
Screenshot 2025-02-04 at 9 31 16 PM

**Review by marionnegp — CHANGES_REQUESTED** [review]
- [x] Should be there be space between the content and the close button? (cc @dmazon-redhat)
Screenshot 2025-02-05 at 4 01 24 PM

- [x] I noticed that the dark color palettes all have the overriding `dark-alt` surface color. Should the light color palettes all use the `lightest` surface color?
Screenshot 2025-02-05 at 3 50 11 PM

**Comment by adamjohnson** [comment]
Made some spacing adjustments. PTAL.

> I noticed that the dark color palettes all have the overriding dark-alt surface color. Should the light color palettes all use the lightest surface color?

@dmazon-redhat let me know if this change needs made. TY!

**Comment by adamjohnson** [comment]
FYI [docs](https://deploy-preview-2153--red-hat-design-system.netlify.app/elements/announcement/) are now up. PTAL.

I have the personalization/implementation docs coded but stashed on my local computer while we confirm they are okay to make public.

File: .changeset/proud-trees-exercise.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+```html
+<rh-announcement>
+  <img slot="image"
+       alt="summit logo"
+       src="/images/summit.png">
+  <p>Lorem ipsum dolor sit amet consectetur adipisicing elit adipisicing elit adipisicing elit.</p>
+  <rh-cta slot="cta">
+    <a href="#">Learn More</a>
+  </rh-cta>
```
Comment (bennypowers) [inline .changeset/proud-trees-exercise.md:18]:
```suggestion
  <rh-cta slot="cta" href="#">Learn More</rh-cta>
```

File: elements/rh-announcement/README.md [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+  <p>Lorem ipsum dolor sit amet consectetur adipisicing elit.</p>
+  <rh-cta slot="cta">
+    <a href="#">Learn More</a>
+  </rh-cta>
+</rh-announcement>
+```
+
+### Announcement Banner using attributes
+- dismissable
+- dark color pallette
```
Comment (bennypowers) [inline elements/rh-announcement/README.md:46]:
```suggestion
- dark color palette
```

maybe this 2-item ul should be prose instead

File: elements/rh-announcement/demo/color-context.html [hunk 3]
```diff
@@ -0,0 +1,35 @@
+<rh-context-demo>
+  <rh-announcement>
+    <svg slot="image"
+         width="80"
+         height="48"
+         role="img"
+         aria-label="Sample image">
+      <rect fill="var(--rh-color-gray-60, #4D4D4D)"
+        stroke="var(--rh-color-border-subtle-on-light, #c7c7c7)"
```
Comment (bennypowers) [inline elements/rh-announcement/demo/color-context.html:9]:
```suggestion
        stroke="var(--rh-color-border-subtle)"
```

File: elements/rh-announcement/demo/color-context.html [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+            y="30"
+            style="font-family: var(--rh-font-family-code, RedHatMono, 'Red Hat Mono', 'Courier New', Courier, monospace); font-size: var(--rh-font-size-body-text-md, 1rem);"
+            fill="var(--rh-color-text-secondary-on-dark, #c7c7c7)">
+              Image
+      </text>
+    </svg>
+    <p>Lorem ipsum dolor sit amet consectetur adipisicing elit adipisicing elit adipisicing elit.</p>
+    <rh-cta slot="cta">
+      <a href="#">Learn More</a>
+    </rh-cta>
```
Comment (bennypowers) [inline elements/rh-announcement/demo/color-context.html:25]:
```suggestion
    <rh-cta slot="cta" href="#">Learn More</rh-cta>
```

File: elements/rh-announcement/README.md [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+
+```html
+<rh-announcement dismissable color-palette="dark">
+  <img slot="image"
+       alt="summit logo"
+       src="/images/summit.png">
+  <p>Lorem ipsum dolor sit amet consectetur adipisicing elit.</p>
+  <rh-cta slot="cta">
+    <a href="#">Learn More</a>
+  </rh-cta>
```
Comment (bennypowers) [inline elements/rh-announcement/README.md:56]:
```suggestion
  <rh-cta slot="cta" href="#">Learn More</rh-cta>
```

File: elements/rh-announcement/README.md [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+
+```html
+<rh-announcement>
+  <img slot="image"
+       alt="summit logo"
+       src="/images/summit.png">
+  <p>Lorem ipsum dolor sit amet consectetur adipisicing elit.</p>
+  <rh-cta slot="cta">
+    <a href="#">Learn More</a>
+  </rh-cta>
```
Comment (bennypowers) [inline elements/rh-announcement/README.md:40]:
```suggestion
  <rh-cta slot="cta" href="#">Learn More</rh-cta>
```

File: elements/rh-announcement/demo/dismissible.html [hunk 7]
```diff
@@ -0,0 +1,86 @@
+<rh-announcement dismissable>
+  <svg slot="image"
+       width="80"
+       height="48"
+       role="img"
+       aria-label="Sample image">
+    <rect fill="var(--rh-color-border-interactive-on-light, #0066cc)"
```
Comment (bennypowers) [inline elements/rh-announcement/demo/dismissible.html:7]:
```suggestion
    <rect fill="var(--rh-color-border-interactive)"
```

File: elements/rh-announcement/demo/dismissible.html [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+      fill-opacity="0.1"
+      stroke="var(--rh-color-border-interactive-on-light, #0066cc)"
```
Comment (bennypowers) [inline elements/rh-announcement/demo/dismissible.html:9]:
```suggestion
      stroke="var(--rh-color-border-interactive)"
```

File: elements/rh-announcement/demo/dismissible.html [hunk 9]
```diff
@@ -0,0 +1,86 @@
...
+</rh-announcement>
+
+<rh-announcement dismissable color-palette="dark" image-position="inline-start">
+  <svg slot="image"
+       width="80"
+       height="48"
+       role="img"
+       aria-label="Sample image">
+    <rect fill="var(--rh-color-gray-60, #4D4D4D)"
+      stroke="var(--rh-color-border-subtle-on-light, #c7c7c7)"
```
Comment (bennypowers) [inline elements/rh-announcement/demo/dismissible.html:35]:
```suggestion
      stroke="var(--rh-color-border-subtle)"
```

File: elements/rh-announcement/demo/dismissible.html [hunk 10]
```diff
@@ -0,0 +1,86 @@
...
+          y="30"
+          style="font-family: var(--rh-font-family-code, RedHatMono, 'Red Hat Mono', 'Courier New', Courier, monospace); font-size: var(--rh-font-size-body-text-md, 1rem);"
+          fill="var(--rh-color-blue-50, #0066CC)">
+            Image
+    </text>
+  </svg>
+  <p>Lorem ipsum dolor sit amet consectetur adipisicing elit ipsum dolor sit.</p>
+  <rh-cta slot="cta">
+      <a href="#">Learn More</a>
+  </rh-cta>
```
Comment (bennypowers) [inline elements/rh-announcement/demo/dismissible.html:25]:
```suggestion
  <rh-cta slot="cta" href="#">Learn More</rh-cta>
```

File: elements/rh-announcement/demo/dismissible.html [hunk 11]
```diff
@@ -0,0 +1,86 @@
...
+      stroke="var(--rh-color-border-subtle-on-light, #c7c7c7)"
+      stroke-width="2px"
+      width="100%"
+      height="100%"
+      stroke-dasharray="4 4">
+    </rect>
+    <text x="17"
+          y="30"
+          style="font-family: var(--rh-font-family-code, RedHatMono, 'Red Hat Mono', 'Courier New', Courier, monospace); font-size: var(--rh-font-size-body-text-md, 1rem);"
+          fill="var(--rh-color-text-secondary-on-dark, #c7c7c7)">
```
Comment (bennypowers) [inline elements/rh-announcement/demo/dismissible.html:44]:
```suggestion
          fill="var(--rh-color-text-secondary)">
```

File: elements/rh-announcement/demo/dismissible.html [hunk 12]
```diff
@@ -0,0 +1,86 @@
...
+            Image
+    </text>
+  </svg>
+  <p>An announcement with <code>inline-start</code> for the value of <code>image position</code>.</p>
+  <rh-cta slot="cta">
+      <a href="#">Learn More</a>
+  </rh-cta>
```
Comment (bennypowers) [inline elements/rh-announcement/demo/dismissible.html:51]:
```suggestion
  <rh-cta slot="cta" href="#">Learn More</rh-cta>
```

File: elements/rh-announcement/demo/dismissible.html [hunk 13]
```diff
@@ -0,0 +1,86 @@
...
+</rh-announcement>
+
+<rh-announcement dismissable image-position="block-start">
+  <svg slot="image"
+       width="80"
+       height="48"
+       role="img"
+       aria-label="Sample image">
+    <rect fill="var(--rh-color-border-interactive-on-light, #0066cc)"
```
Comment (bennypowers) [inline elements/rh-announcement/demo/dismissible.html:60]:
```suggestion
    <rect fill="var(--rh-color-border-interactive)"
```

File: elements/rh-announcement/demo/dismissible.html [hunk 14]
```diff
@@ -0,0 +1,86 @@
...
+      fill-opacity="0.1"
+      stroke="var(--rh-color-border-interactive-on-light, #0066cc)"
```
Comment (bennypowers) [inline elements/rh-announcement/demo/dismissible.html:62]:
```suggestion
      stroke="var(--rh-color-border-interactive)"
```

File: elements/rh-announcement/demo/events.html [hunk 15]
```diff
@@ -0,0 +1,41 @@
+<rh-announcement dismissable>
+  <svg slot="image"
+       width="80"
+       height="48"
+       role="img"
+       aria-label="Sample image">
+    <rect fill="var(--rh-color-border-interactive-on-light, #0066cc)"
```
Comment (bennypowers) [inline elements/rh-announcement/demo/events.html:7]:
```suggestion
    <rect fill="var(--rh-color-border-interactive)"
```

File: elements/rh-announcement/demo/events.html [hunk 16]
```diff
@@ -0,0 +1,41 @@
...
+      fill-opacity="0.1"
+      stroke="var(--rh-color-border-interactive-on-light, #0066cc)"
```
Comment (bennypowers) [inline elements/rh-announcement/demo/events.html:9]:
```suggestion
      stroke="var(--rh-color-border-interactive)"
```

File: elements/rh-announcement/demo/dismissible.html [hunk 17]
```diff
@@ -0,0 +1,86 @@
...
+          y="30"
+          style="font-family: var(--rh-font-family-code, RedHatMono, 'Red Hat Mono', 'Courier New', Courier, monospace); font-size: var(--rh-font-size-body-text-md, 1rem);"
+          fill="var(--rh-color-blue-50, #0066CC)">
+            Image
+    </text>
+  </svg>
+  <p>An announcement with <code>block-start</code> for the value of <code>image position</code>.</p>
+  <rh-cta slot="cta">
+      <a href="#">Learn More</a>
+  </rh-cta>
```
Comment (bennypowers) [inline elements/rh-announcement/demo/dismissible.html:78]:
```suggestion
  <rh-cta slot="cta" href="#">Learn More</rh-cta>
```

File: elements/rh-announcement/demo/events.html [hunk 18]
```diff
@@ -0,0 +1,41 @@
...
+          y="30"
+          style="font-family: var(--rh-font-family-code, RedHatMono, 'Red Hat Mono', 'Courier New', Courier, monospace); font-size: var(--rh-font-size-body-text-md, 1rem);"
+          fill="var(--rh-color-blue-50, #0066CC)">
+            Image
+    </text>
+  </svg>
+  <p>Click the close button to fire the <code>close</code> event.</p>
+  <rh-cta slot="cta">
+    <a href="#">Learn More</a>
+  </rh-cta>
```
Comment (bennypowers) [inline elements/rh-announcement/demo/events.html:25]:
```suggestion
  <rh-cta slot="cta" href="#">Learn More</rh-cta>
```

File: elements/rh-announcement/demo/image-position.html [hunk 19]
```diff
@@ -0,0 +1,59 @@
+<rh-announcement image-position="inline-start">
+  <svg slot="image"
+       width="80"
+       height="48"
+       role="img"
+       aria-label="Sample image">
+    <rect fill="var(--rh-color-border-interactive-on-light, #0066cc)"
```
Comment (bennypowers) [inline elements/rh-announcement/demo/image-position.html:7]:
```suggestion
    <rect fill="var(--rh-color-border-interactive)"
```

File: elements/rh-announcement/demo/image-position.html [hunk 20]
```diff
@@ -0,0 +1,59 @@
...
+      fill-opacity="0.1"
+      stroke="var(--rh-color-border-interactive-on-light, #0066cc)"
```
Comment (bennypowers) [inline elements/rh-announcement/demo/image-position.html:9]:
```suggestion
      stroke="var(--rh-color-border-interactive)"
```

File: elements/rh-announcement/demo/image-position.html [hunk 21]
```diff
@@ -0,0 +1,59 @@
...
+          y="30"
+          style="font-family: var(--rh-font-family-code, RedHatMono, 'Red Hat Mono', 'Courier New', Courier, monospace); font-size: var(--rh-font-size-body-text-md, 1rem);"
+          fill="var(--rh-color-blue-50, #0066CC)">
+            Image
+    </text>
+  </svg>
+  <p>On mobile viewports, the image will stay to the left (inline-start) of this main body content.</p>
+  <rh-cta slot="cta">
+      <a href="#">Learn More</a>
+  </rh-cta>
```
Comment (bennypowers) [inline elements/rh-announcement/demo/image-position.html:25]:
```suggestion
  <rh-cta slot="cta" href="#">Learn More</rh-cta>
```

File: elements/rh-announcement/demo/image-position.html [hunk 22]
```diff
@@ -0,0 +1,59 @@
...
+</rh-announcement>
+
+<rh-announcement image-position="block-start" color-palette="dark">
+  <svg slot="image"
+       width="80"
+       height="48"
+       role="img"
+       aria-label="Sample image">
+    <rect fill="var(--rh-color-gray-60, #4D4D4D)"
+      stroke="var(--rh-color-border-subtle-on-light, #c7c7c7)"
```
Comment (bennypowers) [inline elements/rh-announcement/demo/image-position.html:35]:
```suggestion
      stroke="var(--rh-color-border-subtle)"
```

File: elements/rh-announcement/demo/image-position.html [hunk 23]
```diff
@@ -0,0 +1,59 @@
...
+      stroke-width="2px"
+      width="100%"
+      height="100%"
+      stroke-dasharray="4 4">
+    </rect>
+    <text x="17"
+          y="30"
+          style="font-family: var(--rh-font-family-code, RedHatMono, 'Red Hat Mono', 'Courier New', Courier, monospace); font-size: var(--rh-font-size-body-text-md, 1rem);"
+          fill="var(--rh-color-text-secondary-on-dark, #c7c7c7)">
```
Comment (bennypowers) [inline elements/rh-announcement/demo/image-position.html:44]:
```suggestion
          fill="var(--rh-color-text-secondary)">
```

File: elements/rh-announcement/demo/image-position.html [hunk 24]
```diff
@@ -0,0 +1,59 @@
...
+            Image
+    </text>
+  </svg>
+  <p>In this <code>block-start</code> version, the image stays on top of the body content on mobile viewports.</p>
+  <rh-cta slot="cta">
+      <a href="#">Learn More</a>
+  </rh-cta>
```
Comment (bennypowers) [inline elements/rh-announcement/demo/image-position.html:51]:
```suggestion
  <rh-cta slot="cta" href="#">Learn More</rh-cta>
```

File: elements/rh-announcement/demo/rh-announcement.html [hunk 25]
```diff
@@ -0,0 +1,33 @@
+<rh-announcement>
+  <svg slot="image"
+       width="80"
+       height="48"
+       role="img"
+       aria-label="Sample image">
+    <rect fill="var(--rh-color-border-interactive-on-light, #0066cc)"
```
Comment (bennypowers) [inline elements/rh-announcement/demo/rh-announcement.html:7]:
```suggestion
    <rect fill="var(--rh-color-border-interactive)"
```

File: elements/rh-announcement/demo/rh-announcement.html [hunk 26]
```diff
@@ -0,0 +1,33 @@
...
+      fill-opacity="0.1"
+      stroke="var(--rh-color-border-interactive-on-light, #0066cc)"
```
Comment (bennypowers) [inline elements/rh-announcement/demo/rh-announcement.html:9]:
```suggestion
      stroke="var(--rh-color-border-interactive)"
```

File: elements/rh-announcement/demo/rh-announcement.html [hunk 27]
```diff
@@ -0,0 +1,33 @@
...
+          y="30"
+          style="font-family: var(--rh-font-family-code, RedHatMono, 'Red Hat Mono', 'Courier New', Courier, monospace); font-size: var(--rh-font-size-body-text-md, 1rem);"
+          fill="var(--rh-color-blue-50, #0066CC)">
+            Image
+    </text>
+  </svg>
+  <p>Lorem ipsum dolor sit amet consectetur adipisicing elit adipisicing elit adipisicing elit.</p>
+  <rh-cta slot="cta">
+    <a href="#">Learn More</a>
+  </rh-cta>
```
Comment (bennypowers) [inline elements/rh-announcement/demo/rh-announcement.html:25]:
```suggestion
  <rh-cta slot="cta" href="#">Learn More</rh-cta>
```

File: elements/rh-announcement/rh-announcement-lightdom.css [hunk 28]
```diff
@@ -0,0 +1,94 @@
+rh-announcement:not(:defined) {
```
Comment (bennypowers) [inline elements/rh-announcement/rh-announcement-lightdom.css:1]: Since everything here is `:not(:defined)` perhaps this should be a shim file?

File: elements/rh-announcement/rh-announcement-lightdom.css [hunk 29]
```diff
@@ -0,0 +1,94 @@
...
+  background-color: var(--rh-color-surface-dark-alt, #292929);
+}
+
+rh-announcement[color-palette*='dark']:not(:defined) p {
+  color: var(--rh-color-white, #ffffff);
+}
+
+rh-announcement[color-palette*='dark']:not(:defined) a {
+  color: var(--rh-color-interactive-primary-default-on-dark, #92c5f9);
+}
```
Comment (bennypowers) [inline elements/rh-announcement/rh-announcement-lightdom.css:31]: these two are probably superfluous, should be set by theme css

File: elements/rh-announcement/rh-announcement.css [hunk 30]
```diff
@@ -0,0 +1,151 @@
...
+  display: grid;
+  font-family: var(--rh-font-family-body-text, RedHatText, 'Red Hat Text', Helvetica, Arial, sans-serif);
+  grid-template-columns: 1fr auto 1fr;
+  inline-size: 100%;
+  line-height: var(--rh-line-height-body-text, 1.5);
+  padding: var(--rh-space-lg, 16px);
+  text-align: center;
+
+  &.dark {
+    --rh-color-surface: var(--rh-color-surface-dark-alt, #292929);
```
Comment (bennypowers) [inline elements/rh-announcement/rh-announcement.css:57]: can we avoid `dark-alt`? it's kind of a footgun in our theming system

File: elements/rh-announcement/test/rh-announcement.spec.ts
Comment (bennypowers) [inline elements/rh-announcement/test/rh-announcement.spec.ts:1]: must test close functionality here

**Review by bennypowers — CHANGES_REQUESTED** [review]
overall looks very clean. almost all my comments here are nits, but please do expand on the tests and see if we can remove `dark-alt`

File: elements/rh-announcement/rh-announcement-lightdom.css (hunk 28 above)
Comment (adamjohnson) [reply elements/rh-announcement/rh-announcement-lightdom.css]: Good call, done in [51dbcca](https://github.com/RedHat-UX/red-hat-design-system/pull/2153/commits/51dbcca5efa8cf77bc70cc5f61d5b8006998eb42). 👍

File: elements/rh-announcement/rh-announcement-lightdom.css (hunk 29 above)
Comment (adamjohnson) [reply elements/rh-announcement/rh-announcement-lightdom.css]: Removed these, but in cases where an announcement is dark and the JS fails to load, we'll be leaning on implementors to handle these styles. Something tells me they might not think to test these edge cases. That said, when it happens, they'll certainly see the issue. 😅

File: elements/rh-announcement/test/rh-announcement.spec.ts
Comment (adamjohnson) [reply elements/rh-announcement/test/rh-announcement.spec.ts:1]: Tests added in [e672df0](https://github.com/RedHat-UX/red-hat-design-system/pull/2153/commits/e672df000df89d47885674a470d924753d665965).

File: elements/rh-announcement/rh-announcement.css (hunk 30 above)
Comment (adamjohnson) [reply elements/rh-announcement/rh-announcement.css:57]: This question is likely best answered by @dmazon-redhat. Is it possible to have announcement follow our theming system's colors as follows?

   * Dark 👉 `var(--rh-color-surface-dark, #383838)`
   * Darker 👉 `var(--rh-color-surface-darker, #1f1f1f)`
   * Darkest 👉 `var(--rh-color-surface-darkest, #151515)`

See [this video](https://github.com/user-attachments/assets/ccb3b90b-5027-46e2-8200-d1b29a514344) demonstrating what those colors look like in action.

Most (all?) of our components follow these conventions. If necessary, users can customize this via something like:

```css
rh-announcement[color-palette="darker"] { --rh-color-surface: powderblue; }
```

@bennypowers LMK if you have ideas/direction about this via color transformations (`color-mix`?).

Comment (bennypowers) [reply elements/rh-announcement/rh-announcement.css:57]: maybe something like this: 
```css
#container {
  background: var(--rh-color-surface);
  &.dark {
    background: oklch(from var(--rh-color-surface) calc(l + 10%) c h);
  }
}
```

Comment (adamjohnson) [reply elements/rh-announcement/rh-announcement.css]: Dustin indicated sticking with the default surface colors works for him via chat/slack. This is resolved in [81cc43c](https://github.com/RedHat-UX/red-hat-design-system/pull/2153/commits/81cc43c10778733ba6b121a20191828e02f05cfb).

**Comment by adamjohnson** [comment]
FYI Aly has confirmed with Jeremy and JVP that the documentation is fine as-is—without the implementation/personalization info. I added a bullet to the [Overview page](https://deploy-preview-2153--red-hat-design-system.netlify.app/elements/announcement/#when-to-use) saying that this should only be implemented under the personalization program at their request.

Dustin also confirmed that our normal surface colors are fine. `dark-alt` has been removed.

File: elements/rh-announcement/test/rh-announcement.spec.ts [hunk 31]
```diff
@@ -0,0 +1,82 @@
...
+      const klass = customElements.get('rh-announcement');
+      expect(element)
+          .to.be.an.instanceOf(klass)
+          .and
+          .to.be.an.instanceOf(RhAnnouncement);
+    });
+  });
+});
+
+describe('when the element loads', function() {
```
Comment (bennypowers) [inline elements/rh-announcement/test/rh-announcement.spec.ts:24]:
```suggestion
describe('with `dismissable` and `image-position` attributes, a slotted image, and a slotted cta', function() {
```

File: elements/rh-announcement/test/rh-announcement.spec.ts [hunk 32]
```diff
@@ -0,0 +1,82 @@
...
+
+
+  it('should remove the rh-announcement from DOM after close button click', async function() {
+    const parent = element.parentElement;
+
+    elementCloseButton?.click();
+    await element.updateComplete;
+
+    expect(parent?.contains(element)).to.be.false;
+  });
```
Comment (bennypowers) [inline elements/rh-announcement/test/rh-announcement.spec.ts:81]: this test and the next are fine, but in general we should prefer to nest actions in describe blocks, and keep it blocks for assertions only
```suggestion
  describe('clicking the close button', function() {
    let event: Event
    beforeEach(async function() {
      const eventPromise = oneEvent(element, 'close');
      elementCloseButton?.click();
      event = await eventPromise;
    });
    
    it('should send a close event', async function() {
      expect(event).to.be.an.instanceOf(Event);
      expect(event.type).to.equal('close');
      expect(element.isConnected).to.be.false;
    });
   
    it('should remove the rh-announcement from DOM', async function() {
      expect(element.isConnected).to.be.false;
    });
  });
```

Comment (adamjohnson) [reply elements/rh-announcement/test/rh-announcement.spec.ts]: Great suggestions! Resolved in [58bec2f](https://github.com/RedHat-UX/red-hat-design-system/pull/2153/commits/58bec2ffaba82fa7b696bad8673ef688b2980c60).

File: elements/rh-announcement/test/rh-announcement.spec.ts (hunk 31 above)
Comment (adamjohnson) [reply elements/rh-announcement/test/rh-announcement.spec.ts]: Resolved in [58bec2f](https://github.com/RedHat-UX/red-hat-design-system/pull/2153/commits/58bec2ffaba82fa7b696bad8673ef688b2980c60).

**Comment by adamjohnson** [comment]
Dustin and Aly have confirmed they are ✅ on this and have no further changes or requests. 

This should be ready to merge whenever we are ready.

**Review by bennypowers — CHANGES_REQUESTED** [review]
so looks like we have a spelling consistency problem in our repo: *dismissible* vs *dismissable*

To the best of my knowledge, they're both acceptable spellings. Some claim there is a difference in shades of meaning or formal application but I think that's poppycock.

I would say that since alert already uses `dismissable`, that pretty much requires us to follow suit here.

I also vaguely recall having a similar discussion regarding a similar adjective, maybe 6mo or a year ago? @coreyvickery does that ring a bell?

```
red-hat-design-system on  main [$] is 📦 v2.1.1 via  v23.2.0 
❯ rg dismissible
elements/rh-alert/docs/20-guidelines.md
31:Depending on the message, an inline alert can be dismissible or not dismissible. The close button should not be included if it is critical that a user read or interact with the alert. A toast alert is always dismissible.
34:  <img src="../alert-guidelines-dismissible-inline.svg" 
41:  <img src="../alert-guidelines-dismissible-toast.svg" 
```

```
red-hat-design-system on  main [$] is 📦 v2.1.1 via  v23.2.0 
❯ rg dismissable
docs/accessibility/development.md
119:- Ensure that content that appears on hover or focus is dismissable, hoverable, and persistent for keyboard and mouse users. ([1.4.13 Content on Hover or Focus - Level AA][wcag22u1413])
137:- Ensure content that appears or disappears on changes to mouse hover or keyboard focus is dismissable, hoverable, and persistent. ([1.4.13 Content on Hover or Focus - Level AA][wcag22u1413])

elements/rh-alert/rh-alert.ts
47: * @fires {AlertCloseEvent} close - when the dismissable alert closes
180:  @property({ reflect: true, type: Boolean }) dismissable = false;
243:            </div>${!this.dismissable && this.variant !== 'toast' ? '' : html`

elements/rh-alert/README.md
20:`dismissable`
21:- The dismissable attribute adds a close button to the top right of the alert allowing the user to dismiss the alert. Clicking the close button dispatches a `close` event, then removes the alert from the page.
26:- This is fired when the user clicks the close button enabled by the `dismissable` attribute. Cancelling the event prevents the alert from closing.

elements/rh-alert/test/rh-alert.spec.ts
15:const dismissableTemplate = html`
16:  <rh-alert dismissable>
17:    <h3 slot="header">Default dismissable</h3>
28:  let dismissableElement: RhAlert;
32:    dismissableElement = await fixture<RhAlert>(dismissableTemplate);
43:  describe('dismissable <rh-alert>', async () => {
47:      elementCloseButton = dismissableElement.shadowRoot?.querySelector('#close-button');
50:    it('should only show the close button if the dismissable attribute is present', () => {
55:    it('should send a close event on dismissable close button click', async () => {
56:      elementCloseButton = dismissableElement.shadowRoot?.querySelector('#close-button');
61:      await oneEvent(dismissableElement, 'close');
62:      dismissableElement.requestUpdate();
64:      await dismissableElement.updateComplete;
65:      expect(dismissableElement.isConnected).to.be.false;
69:      elementCloseButton = dismissableElement.shadowRoot?.querySelector('#close-button');
71:      dismissableElement?.addEventListener('close', event => {
77:      await oneEvent(dismissableElement, 'close');
78:      dismissableElement.requestUpdate();
80:      await dismissableElement.updateComplete;
81:      expect(dismissableElement.isConnected).to.be.true;

elements/rh-alert/demo/dismissable.html
1:<section id="alert-dismissable">
2:  <rh-alert dismissable>
3:    <h3 slot="header">Default dismissable</h3>
10:  <rh-alert variant="alternate" dismissable>
11:    <h3 slot="header">Inline dismissable</h3>
18:  <rh-alert toast dismissable>
19:    <h3 slot="header">Toast dismissable</h3>
26:  <rh-alert dismissable data-on-close="prevent-default">
52:  #alert-dismissable {
```

**Review by bennypowers — APPROVED** [review]
Letters got to match!
